jenkins/bootstrap_test.py

```diff
@@ -378,6 +378,42 @@ def tearDown(self):
             with stub:
                 pass
 
+    def testNoVersion(self):
+        gsutil = FakeGSUtil()
+        paths = FakePath()
+        success = True
+        artifacts = 'not-a-dir'
+        no_version = ''
+        version = 'should not have found it'
+        with Stub(bootstrap, 'metadata', lambda *a: {'random-meta': version}):
+            bootstrap.finish(gsutil, paths, success, artifacts, BUILD, no_version, REPO)
+        bootstrap.finish(gsutil, paths, success, artifacts, BUILD, no_version, REPO)
+        calls = gsutil.jsons[-1]
+        # Meta is second positional argument
```

Comment:
What I mean is that the json metadata (what you are calling data) is in the second positional argument (positional arguments are calls[0], the second positional argument is calls[*][1]).